web/src/components/SidebarLink.js

```diff
@@ -24,12 +24,24 @@ const getClasses = (depth, hash, anchor) => deline`
   ${depth ? 'pl2' : ''} ${hash === anchor ? 'sb-item-active' : ''}
 `;
 
+const clickHandler = (depth, toggleExpand, propsOnClick) => e => {
+  if (depth === 0) {
+    toggleExpand();
+  }
+  if (propsOnClick) {
+    propsOnClick();
+    e.preventDefault();
+    e.stopPropagation();
+  }
+};
```

Comment:
This `depth` piece already existed, but the gist of it is that if this link is at the top of the tree, then a click should expand it to show the links underneath it.

The `propsOnClick` thing is new - the idea is that if the thing creating this link hooks up its own click handler, we'll use that and suppress the default click event.